lib/spack/spack/filesystem_view.py

```diff
@@ -67,7 +67,11 @@ def view_copy(src, dst, view, spec=None):
 
     Use spec and view to generate relocations
     """
-    shutil.copy2(src, dst)
+    if os.path.isdir(src):
+        shutil.copytree(src, dst)
+    else:
+        shutil.copy2(src, dst)
+
```

Comment:
`isdir` is true if and only if `src` is a symlink to a dir. You should create a symlink in `dst`.

Unfortunately this is complicated by two things:
1) Python's sad filesystem API does not have a Python 2 compatible `shutil.copy2(src, dst, follow_symlinks=False)`.
2) Symlinks can be absolute paths too, and those need to be relocated as well.

Reply:
How would shutil.copytree symlink=True dare for that case? I can't recall how copy2 differs.